packages/smart-contracts/test/contracts/ERC20SwapToConversion.test.ts

```diff
@@ -273,4 +299,23 @@ describe('contract: ERC20SwapToConversion', () => {
     ).to.be.revertedWith('Could not transfer payment token from swapper-payer');
     await expectPayerBalanceUnchanged();
   });
+
+  it('cannot swap with a bad proxy address', async function () {
+    // Simulate request payment for 10 (fiat) + 1 (fiat) fee, in paymentNetworkErc20
+    await expect(
+      swapConversionProxy.swapTransferWithReference(
+        chainlinkConversion.address, // random address that is not a conversion proxy
+        to,
+        fiatDecimal.mul(10),
+        erc20Decimal.mul(70),
+        [spentErc20.address, paymentNetworkErc20.address], // _uniswapPath
+        [USDhash, paymentNetworkErc20.address], // _chainlinkPath
+        referenceExample,
+        fiatDecimal.mul(1),
+        builder,
+        exchangeRateOrigin + 1000, // _uniswapDeadline. 100 -> 1000: Too low value may lead to error (network dependent)
+        0, // _chainlinkMaxRateTimespan
+      ),
+    ).to.be.revertedWith('');
```

Comment:
Don't you have a revert message to test with more precision?

Reply:
When we do `paymentProxy.transferFromWithReferenceAndFee` the function does not exists and the tx just reverts.